puresnmp/__init__.py

```diff
@@ -22,101 +22,105 @@
 from .transport import send, get_request_id
 
 
-def get(ip: str, community: str, oid: str, version: bytes=Version.V2C,
-        port: int=161):
+class SnmpClient(object):
     """
-    Executes a simple SNMP GET request and returns a pure Python data structure.
+    SNMP client object used to send SNMP requests to target devices.
     """
 
-    oid = ObjectIdentifier.from_string(oid)
+    def __init__(self, community: str="public", version: bytes=Version.V2C, port: int=161, timeout: int=2):
```

Comment:
I'm a bit iffy about passing all these arguments to the class's `__init__`. You could imagine that you can access the same device/ip with different communities, given existing authorisation rules. For example you might use one community for read access, and another for write access.

I think that, if doing something like this, either the `ip` should be moved into `__init__` as well, or, that all arguments except `timeout` should be pushed down to the individual functions again.

Both approaches make sense. It's just that the user experience will be different. In one case you have one `SnmpClient` instance per connection, in the other case you have one SnmpClient instance per application.

I personally prefer the second option. I think it's weird to instantiate a new class every time you want to make a  call.

I think the proposal to simply pass a `Transport` instance into the `SnmpClient` class and leave everything else as-is, might come out the cleanest.